src/NuGet.Core/NuGet.Protocol.Core.v3/RemoteRepositories/PackageUtilities.cs

```diff
@@ -25,8 +25,9 @@ internal static class PackageUtilities
                 try
                 {
                     using (var reader = new PackageArchiveReader(nupkgStream, leaveStreamOpen: true))
+                    using (var nuspecStream = reader.GetNuspec())
                     {
-                        return new NuspecReader(reader.GetNuspec());
+                        return new NuspecReader(nuspecStream);
                     }
```

Comment:
Localize the string below while you are at it report.LogWarning($"The ZIP archive {fileStream.Name} is corrupt");